client/my-sites/upgrades/map-domain/index.jsx

```diff
@@ -64,11 +64,18 @@ var MapDomain = React.createClass( {
 	},
 
 	goBack: function() {
-		if ( ! this.props.sites ) {
-			return page( this.props.path.replace( '/mapping', '' ) );
+		const selectedSite = this.props.sites && this.props.sites.getSelectedSite();
+		if ( ! selectedSite ) {
+			page( this.props.path.replace( selectedSite.is_vip ? '/add/mapping' : '/mapping', '' ) );
+			return;
```

Comment:
Thanks, great catch! We can actually skip any additional logic here, since the `upgradesController.redirectToAddMappingIfVipSite` function will redirect if needed anyway.